File: frontend/src/lib/audit-log.ts
import { api } from "./api";

export interface AuditLogItem {
  id: string;
  timestamp: string;
  usuario_id: string | null;
  usuario_email: string | null;
  acao: string;
  entidade: string | null;
  entidade_id: string | null;
  diff_json: Record<string, unknown> | null;
}

export interface AuditLogPage {
  items: AuditLogItem[];
  next_cursor: string | null;
}

export interface AuditLogParams {
  acao?: string;
  cursor?: string;
}

export async function listAuditLog(params: AuditLogParams = {}): Promise<AuditLogPage> {
  const { data } = await api.get<AuditLogPage>("/v1/audit-log", { params });
  return data;
}

File: backend/finacialsim_saas/schemas/audit_log.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class AuditLogItem(BaseModel):
    id: uuid.UUID
    tenant_id: uuid.UUID
    timestamp: datetime
    usuario_id: uuid.UUID | None
    usuario_email: str | None = None
    acao: str
    entidade: str | None
    entidade_id: uuid.UUID | None
    diff_json: dict | None

    model_config = {"from_attributes": True}


class AuditLogPage(BaseModel):
    items: list[AuditLogItem]
    next_cursor: str | None
